server/chat-plugins/blackjack.js

```diff
@@ -0,0 +1,643 @@
+		output += `<br /><button class="button" name="send" value="/blackjack hit" title="Hit (get another card)">Hit</button> | <button class="button" name="send" value="/blackjack stand" title="Stand (just keep these cards)">Stand</button>`;
+
+		this.playerTable[this.curUser].sendRoom(`|notify|Blackjack (${this.room.title})|It's your turn to play!`);
+		this.playerTable[this.curUser].sendRoom(output);
+		this.playerTable[this.curUser].playScreen = output;
+
+		this.createTimer(this.playerTable[this.curUser]);
+	}
+}
+
+class BlackjackPlayer extends Rooms.RoomGamePlayer {
+	constructor(user, game) {
+		super(user, game);
+		this.game = game;
+
+		this.cards = [];
+		this.points = 0;
+		this.slide = 0;
+		this.status = 'playing';
+
+		this.selfUhtml = '';
+		this.gameLog = '';
+		this.playScreen = '';
+		this.timerTicksLeft = this.game.turnTimeoutMinutes * 60; // to get into seconds-format
+	}
+	resetTimerTicks() {
+		this.timerTicksLeft = this.game.turnTimeoutMinutes * 60;
+	}
+}
+
+class BlackjackDealer {
+	constructor() {
+		this.cards = [];
+		this.points = 0;
+		this.name = 'The Dealer';
+	}
+}
+
+class BlackjackDeck {
+	constructor() {
+		this.deck = [
+			'A♥', 'A♦', 'A♣', 'A♠', '2♥', '2♦', '2♣', '2♠', '3♥', '3♦', '3♣',
+			'3♠', '4♥', '4♦', '4♣', '4♠', '5♥', '5♦', '5♣', '5♠', '6♥', '6♦', '6♣', '6♠',
+			'7♥', '7♦', '7♣', '7♠', '8♥', '8♦', '8♣', '8♠', '9♥', '9♦', '9♣', '9♠', '10♥',
+			'10♦', '10♣', '10♠', 'J♥', 'J♦', 'J♣', 'J♠', 'Q♥', 'Q♦', 'Q♣', 'Q♠', 'K♥', 'K♦',
+			'K♣', 'K♠',
+		];
+	}
+	shuffle() {
+		return Dex.shuffle(this.deck);
+	}
+}
+
+exports.commands = {
+	bj: 'blackjack',
+	blackjack: {
+		new: 'create',
+		create(target, room, user) {
+			if (!this.can('gamemanagement', null, room)) return;
+			if (room.game) return this.errorReply("There is already a game running in this room.");
+			if (room.blackjackDisabled) return this.errorReply("Blackjack is currently disabled in this room.");
+
+			this.privateModAction(`(A game of blackjack was created by ${user.name}.)`);
+			this.modlog(`BLACKJACK CREATE`);
+			room.game = new Blackjack(room, user, target);
+		},
+		start(target, room, user) {
+			if (!this.can('gamemanagement', null, room)) return;
+			if (!room.game || !room.game.blackjack) return this.errorReply("There is no game of blackjack currently ongoing in this room.");
+
+			this.privateModAction(`(The game of blackjack was started by ${user.name}.)`);
+			this.modlog(`BLACKJACK START`);
+			room.game.start(user);
+		},
+		forceend: 'end',
+		end(target, room, user, connection, cmd) {
+			if (!this.can('gamemanagement', null, room)) return;
+			if (!room.game || !room.game.blackjack) return this.errorReply("There is no game of blackjack currently ongoing in this room.");
+
+			this.privateModAction(`The game of blackjack was ended by ${user.name}.`);
+			this.modlog(`BLACKJACK END`);
+			room.game.destroy();
```

Comment:
Do you mean to call `destroy` and not `end` here?

Reply:
Yes

Reply:
You appear to only ever call `Blackjack#end` with no arguments, then, which looks very wrong compared to its signature?

Reply:
(seem to have misinterpreted what you meant, nvm)